src/BloomExe/Publish/PdfMaker.cs

```diff
@@ -71,10 +73,67 @@ public void MakePdf(string inputHtmlPath, string outputPdfPath, string paperSize
 				}));
 			}
 
-			if (bookletPortion != PublishModel.BookletPortions.AllPagesNoBooklet)
+			try
 			{
-				//remake the pdf by reording the pages (and sometimes rotating, shrinking, etc)
-				MakeBooklet(outputPdfPath, paperSizeName, booketLayoutMethod, layoutPagesForRightToLeft);
+				if (bookletPortion != PublishModel.BookletPortions.AllPagesNoBooklet)
+				{
+					//remake the pdf by reording the pages (and sometimes rotating, shrinking, etc)
+					MakeBooklet(outputPdfPath, paperSizeName, booketLayoutMethod, layoutPagesForRightToLeft);
+				}
+				else
+				{
+					 // Just check that we got a valid, readable PDF. (MakeBooklet has to read the PDF itself,
+					// so we don't need to do this check if we're calling that.)
+					// If we get a reliable fix to BL-932 we can take this 'else' out altogether.
+					CheckPdf(outputPdfPath);
+				}
+			}
+			catch (KeyNotFoundException e)
+			{
+				// This is characteristic of BL-932, where Gecko29 fails to make a valid PDF, typically
+				// because the user has embedded a really huge image, something like 4000 pixels wide.
+				// We think it could also happen with a very long book or if the user is short of memory.
+				// The resulting corruption of the PDF file takes the form of a syntax error in an embedded
+				// object so that the parser finds an empty string where it expected a 'generationNumber'
+				// (currently line 106 of Parser.cs). This exception is swallowed but leads to an empty
+				// externalIDs dictionary in PdfImportedObjectTable, and eventually a new exception trying
+				// to look up an object ID at line 121 of that class. We catch that exception here and
+				// suggest possible actions the user can take until we find a better solution.
+				MessageBox.Show(
```

Comment:
If instead you use
     Palaso.Reporting.ErrorReport.NotifyUserOfProblem(error, msg),

Then a user would have buttons to report the problem to us.